File: packages/storage/ios/Plugin/Classes/Options/UpdateMetadataOptions.swift
import Foundation
import FirebaseStorage
import Capacitor

@objc public class UpdateMetadataOptions: NSObject {
    private var path: String
    private var metadata: StorageMetadata

    init(path: String, metadata: JSObject) {
        self.path = path
        self.metadata = FirebaseStorageHelper.buildStorageMetadata(metadata)
    }

    func getPath() -> String {
        return path
    }

    func getMetadata() -> StorageMetadata {
        return metadata
    }
}

File: packages/storage/android/src/main/java/app/capgo/capacitor/firebase/storage/UpdateMetadataOptions.java
package app.capgo.capacitor.firebase.storage.classes.options;

import com.getcapacitor.JSObject;
import com.google.firebase.storage.StorageMetadata;
import app.capgo.capacitor.firebase.storage.FirebaseStorageHelper;
import org.json.JSONException;

public class UpdateMetadataOptions {

    private String path;
    private StorageMetadata metadata;

    public UpdateMetadataOptions(String path, JSObject metadata) throws JSONException {
        this.path = path;
        this.metadata = FirebaseStorageHelper.buildStorageMetadata(metadata);
    }

    public String getPath() {
        return path;
    }

    public StorageMetadata getMetadata() {
        return metadata;
    }
}
